### DISCUSSION TITLE
Dependencies macro tries to initialize static vars

### DISCUSSION THREAD

**Discussion by BastianKusserow** [pointfreeco/swift-dependencies#200] [General]
### Description

Hey!
Seems like the DependencyClient macro tries to initialize static vars within the generated init. 
So the following code
```swift
@DependencyClient
public struct SimpleDependency {
    static var foo: Int = 42
    var bar: () -> Void
}
```
generates
```swift
@DependencyClient
public struct SimpleDependency {
    static var foo: Int = 42
    @DependencyEndpoint
    var bar: () -> Void
    init(
        foo: Int = 42,
        bar: @escaping () -> Void
    ) {
        self.foo = foo
        self.bar = bar
    }
}
```
This does not compile for obvious reasons. Is this just an oversight in the macros implementation or should static vars just not be supported?

### Checklist

- [x] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [x] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-dependencies/issues) or [discussion](https://github.com/pointfreeco/swift-dependencies/discussions).

### Expected behavior

I'd expect that the static var is just ignored in the initializer. However, as I was seeing the issue, I was also searching for something like `@DependencyEndpointIgnored` to somehow mitigate the issue. Maybe thats also something worth exploring?

### Actual behavior

The initializer tries to initialize the static var, which is not possible.

### Swift Compiler version information

```shell
swift-driver version: 1.90.11.1 Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
```

**Comment by mbrandonw** [comment]
Hi @BastianKusserow, this can certainly be fixed, but static vars are not safe to use in Swift 6 (on non-global actors at least), and so this really isn't something you should be doing anyway. If you turn on concurrency checks you will get the following warning:

> ⚠️ Static property 'foo' is not concurrency-safe because it is non-isolated global shared mutable state; this is an error in Swift 6

If you have any interest in fixing we would happily take a PR, but I'm not sure when we will have the time to look into it.

For now I am going to convert this to a discussion since it is not a top priority for us, so please continue any discussion over there.

**Reply by rhysm94** [reply]
I think this is something that was fixed with my PR from a couple of weeks ago now :)

https://github.com/pointfreeco/swift-dependencies/commit/083ff59d1d4fab2965d14fd611e8d2dd6675cdfd
